server/url_params.go

```diff
@@ -11,17 +11,49 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	// Query parameters
+	ParamHint     = "hint"
+	ParamOriginId = "originid"
+	ParamBuilder  = "builder"
+	ParamRefund   = "refund"
+
+	// Maximum length of the originId query parameter
+	MaxOriginIdLength = 255
+
+	// Maximum and minimum values for the refund percentage
+	MaxRefundPercent = 100
+	MinRefundPercent = 0
+)
+
 var (
 	DefaultAuctionHint = []string{"hash", "special_logs"}
 
-	ErrEmptyHintQuery                      = errors.New("Hint query must be non-empty if set.")
-	ErrEmptyTargetBuilderQuery             = errors.New("Target builder query must be non-empty if set.")
-	ErrIncorrectAuctionHints               = errors.New("Incorrect auction hint, must be one of: contract_address, function_selector, logs, calldata, default_logs.")
-	ErrIncorrectOriginId                   = errors.New("Incorrect origin id, must be less then 255 char.")
-	ErrIncorrectRefundQuery                = errors.New("Incorrect refund query, must be 0xaddress:percentage.")
-	ErrIncorrectRefundAddressQuery         = errors.New("Incorrect refund address.")
-	ErrIncorrectRefundPercentageQuery      = errors.New("Incorrect refund percentage.")
-	ErrIncorrectRefundTotalPercentageQuery = errors.New("Incorrect refund total percentage, must be bellow 100%.")
+	validHints = map[string]struct{}{
+		"hash":              {},
+		"contract_address":  {},
+		"function_selector": {},
+		"logs":              {},
+		"calldata":          {},
+		"default_logs":      {},
+	}
+
+	ErrEmptyHintQuery          = errors.New("hint query must be non-empty if set")
+	ErrEmptyTargetBuilderQuery = errors.New("target builder query must be non-empty if set")
+	ErrIncorrectAuctionHints   = func(hint string) error {
+		return fmt.Errorf("incorrect auction hint: %s, must be one of: contract_address, function_selector, logs, calldata, default_logs", hint)
+	}
+	ErrIncorrectOriginId = func(originId string) error {
```

Comment:
This is non-idiomatic way of declaring errors in go due to the fact that you can't leverage `errors.Is` this way. There should be constant underlying `ErrIncorrectOriginID=fmt.Errof("incorrect origin id, must be non-empty and less than 255 characters")`. If you want additional context you use `fmt.Errorf("%w value: %s, ErrIncorrectOriginID, originID)`